src/style/style.js

```diff
@@ -501,6 +502,27 @@ class Style extends Evented {
     }
 
     /**
+    * Set the data of a source, given its id. Only available for GeoJSON sources.
+    * @param {string} id id of the source
+    * @param {GeoJSON|string} data GeoJSON source
+    * @throws {Error} if no source is found with the given ID
+    */
+
+    setData(id: string, data: GeoJSON | string) {
+        this._checkLoaded();
+
+        if (this.sourceCaches[id] === undefined) {
+            throw new Error('There is no source with this ID');
+        }
```

Comment:
Since this is an internal invariant, please replace this check with just `assert(this.sourceCaches[id] === undefined, 'There is no source with this ID')`